app/src/main/java/com/github/sdpteam15/polyevents/helper/HelperFunctions.kt

```diff
@@ -381,4 +383,66 @@ object HelperFunctions {
             }
         builder.show()
     }
+
+    /**
+     * Color.ORANGE does not exist, so we created it here
+     * by lazy is here to prevent the "color is not mocked" error
+     */
+    val ORANGE by lazy{ Color.rgb(255,165,0) }
+
+    /**
+     * Converts any Object into an Integer, if null return null.
+     * Throws an Exception if the type is not a number
+     */
+    fun Any?.toInt() = when(this){
+        is Int -> this
+        is Long -> this.toInt()
+        is Float -> this.toInt()
+        is Double -> this.toInt()
+        null -> null
+        else -> throw TypeCastException("Cannot call toInt on ${this::class.java.name}")
+    }
+    /**
+     * Converts any Object into a Long, if null return null.
+     * Throws an Exception if the type is not a number
+     */
+    fun Any?.toLong() = when(this){
+        is Int -> this.toLong()
+        is Long -> this
+        is Float -> this.toLong()
+        is Double -> this.toLong()
+        null -> null
+        else -> throw TypeCastException("Cannot call toLong on ${this::class.java.name}")
+    }
+    /**
+     * Converts any Object into an Float, if null return null.
+     * Throws an Exception if the type is not a number
+     */
+    fun Any?.toFloat() = when(this){
+        is Int -> this.toFloat()
+        is Long -> this.toFloat()
+        is Float -> this
+        is Double -> this.toFloat()
+        null -> null
+        else -> throw TypeCastException("Cannot call toFloat on ${this::class.java.name}")
+    }
+    /**
+     * Converts any Object into an Double, if null return null.
+     * Throws an Exception if the type is not a number
+     */
+    fun Any?.toDouble() = when(this){
+        is Int -> this.toDouble()
+        is Long -> this.toDouble()
+        is Float -> this.toDouble()
+        is Double -> this
```

Comment:
Can we add the String class here ?

Reply:
This is to avoid using "(document["___"] as Long).toInt()" everywhere in our firestore adapters and to use them later in the local cache layer of the database. I think it would be more appropriate to use Int.parseInt(...) in that case.